app/controllers/labels_controller.rb

```diff
@@ -3,7 +3,21 @@
 class LabelsController < ApplicationController
   load_and_authorize_resource
 
+  before_action :sanitize_page_param, only: :search
   def search
-    @labels = Label.order(:name)
+    results = Label.ransack(name_i_cont: params[:search]).result
+    paginated = results.paginate(per_page: 3, page: @page)
+
+    render json: {
+      results: paginated.map {|l| {id: l.name, text: l.name, label_color: l.color, label_font_color: l.font_color} },
+      pagination: {more: (@page * 3 < results.length)},
```

Comment:
After my previous two suggestion, this line reads as:

```rb
      pagination: {more: (@page * paginated.limit_value < paginated.total_entries)},
```

I feel, this is better, but still not perfect. When checking the library further, I saw the following possibilities:

```rb
      pagination: {more: paginated.next_page.present?},
```

[According to the source](https://github.com/mislav/will_paginate/blob/8113fddf4b506867e9af3aa59fe2df314b79860c/lib/will_paginate/collection.rb#L24-L27), the `next_page` will either return the number of the next page or nil. Both is fine, and we can convert this result to the desired boolean value with `present?`.

Overall, I would go with this solution. I hope you agree that this solution is more elegant and also easier to read :).

Reply:
Yes, that is true.